src/main/java/org/magicdgs/thaplv/haplotypes/model/HaplotypeConverter.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2016, Daniel Gomez-Sanchez <daniel.gomez.sanchez@hotmail All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without modification, are permitted
+ * provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this list of conditions
+ * and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions
+ * and the following disclaimer in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse
+ * or promote products derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+ * THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+package org.magicdgs.thaplv.haplotypes.model;
+
+import htsjdk.variant.variantcontext.VariantContext;
+
+import java.util.function.Function;
+
+/**
+ * Haplotype converter is a functional interface that converts a {@link VariantContext} with
+ * diploid individuals into haploid. {@link VariantContext} with already haploid individuals, a
+ * copy of the variant will be returned.
+ *
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ */
+@FunctionalInterface
+public interface HaplotypeConverter extends Function<VariantContext, VariantContext> {
```

Comment:
Why `HaplotypeConverter`is a `Function<VariantContext, VariantContext>` if all the methods are applied to genotypes, and it just iterates over the genotypes in the `VariantContext`? Probably it will be better to change the abstraction to a `Function<Genotype, Genotype>`, and `getHaplotype()` being the default apply...